From 58a2065fba1387b1b8b069ce2da521217eb8e523 Mon Sep 17 00:00:00 2001
From: Feiyang Chen <chenfeiyang@loongson.cn>
Date: Sat, 6 Jan 2024 15:27:13 +1100
Subject: [PATCH 4/4] Add nightly tests and README for loongarch64

Add nightly tests for loongarch64-linux platform. Add a README for
loongarch64.

Signed-off-by: Feiyang Chen <chenfeiyang@loongson.cn>
Signed-off-by: Shi Pujin <shipujin@loongson.cn>
---
 README                             |  1 +
 README.loongarch64                 | 51 ++++++++++++++++++++++++++++++
 nightly/conf/loongarch64.conf      |  3 ++
 nightly/conf/loongarch64.sendemail |  7 ++++
 4 files changed, 62 insertions(+)
 create mode 100644 README.loongarch64
 create mode 100644 nightly/conf/loongarch64.conf
 create mode 100644 nightly/conf/loongarch64.sendemail

diff --git a/README b/README
index 842388036..1a88d1d47 100644
--- a/README
+++ b/README
@@ -41,6 +41,7 @@ platforms:
 - MIPS32/Linux
 - MIPS64/Linux
 - nanoMIPS/Linux
+- LOONGARCH64/Linux
 - X86/Solaris
 - AMD64/Solaris
 - X86/FreeBSD
diff --git a/README.loongarch64 b/README.loongarch64
new file mode 100644
index 000000000..742bca085
--- /dev/null
+++ b/README.loongarch64
@@ -0,0 +1,51 @@
+Status
+~~~~~~
+
+A port to LoongArch64 Linux platform.
+
+Some new IROps:
+
+* Iop_ScaleBF64
+* Iop_ScaleBF32
+* Iop_RSqrtF64
+* Iop_RSqrtF32
+* Iop_LogBF64
+* Iop_LogBF32
+* Iop_MaxNumAbsF64
+* Iop_MinNumAbsF64
+* Iop_MaxNumF32
+* Iop_MinNumF32
+* Iop_MaxNumAbsF32
+* Iop_MinNumAbsF32
+* Iop_Max64Sx4
+* Iop_Max64Ux4
+* Iop_Min64Sx4
+* Iop_Min64Ux4
+
+A new IRMBusEvent:
+
+* Imbe_InsnFence
+
+A new IRJumpKind:
+
+* Ijk_SigSYS
+
+
+Limitations
+~~~~~~~~~~~
+
+* Only the fallback LLSC implementation is supported.
+* Not all vector instructions are supported.
+* Tests need to be added for vector instructions.
+
+
+Reading Material
+~~~~~~~~~~~~~~~~
+
+* LoongArch Reference Manual
+  https://loongson.github.io/LoongArch-Documentation/LoongArch-Vol1-EN.html
+* LoongArch ELF ABI specification:
+  https://loongson.github.io/LoongArch-Documentation/LoongArch-ELF-ABI-EN.html
+* LoongArch Toolchain Conventions:
+  https://loongson.github.io/LoongArch-Documentation/LoongArch-toolchain-conventions-EN.html
+
diff --git a/nightly/conf/loongarch64.conf b/nightly/conf/loongarch64.conf
new file mode 100644
index 000000000..8603671db
--- /dev/null
+++ b/nightly/conf/loongarch64.conf
@@ -0,0 +1,3 @@
+export ABT_DETAILS=`uname -mrs`
+export ABT_JOBS=4
+export ABT_PERF="--vg=../valgrind-new --vg=../valgrind-old"
diff --git a/nightly/conf/loongarch64.sendemail b/nightly/conf/loongarch64.sendemail
new file mode 100644
index 000000000..d784da34f
--- /dev/null
+++ b/nightly/conf/loongarch64.sendemail
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+subject=$1
+body=$2
+file=$3
+
+(cat "$body" "$file") | /usr/bin/mail -s "$subject" valgrind-testresults@lists.sourceforge.net -r "Feiyang Chen <chris.chenfeiyang@gmail.com>"
-- 
2.45.2

